src/bin/svg2png/svg2png.cpp

```diff
@@ -299,15 +325,42 @@ struct App
         return renderer.render(path, width, height, dst, bgColor);
     }
 
-    int handleDirectory(const string& path, DIR* dir)
+    int handleDirectory(const string& path)
     {
         //open directory
-        if (!dir) {
-            dir = opendir(path.c_str());
-            if (!dir) {
-                cout << "Couldn't open directory \"" << path.c_str() << "\"." << endl;
-                return 1;
+#ifdef _WIN32
+        WIN32_FIND_DATA fd;
+        HANDLE h = FindFirstFileEx((path + "\\*").c_str(), FindExInfoBasic, &fd, FindExSearchNameMatch, NULL, 0);
```

Comment:
just I have a question because I'm curious.
Is there any reason to use `FindFirstFileEx`? (instead of `FindFirstFile`)

Reply:
yes: by default, FindFirstFile() search for files with also the old DOS file names (8 chars for file name + 3 for extension). Nowadays, there is no such limitation on the files. To remove this search, I use FindFirstFileEx() with the option FindExInfoBasic. Hence the search is faster. See this [link](https://docs.microsoft.com/en-us/windows/win32/api/minwinbase/ne-minwinbase-findex_info_levels)

Reply:
notealso that i benchmarked this some years ago to verify this. Indeed it is faster.